numpy/ma/core.py

```diff
@@ -4790,7 +4790,11 @@ def all(self, axis=None, out=None, keepdims=np._NoValue):
 
         mask = _check_mask_axis(self._mask, axis, **kwargs)
         if out is None:
-            d = self.filled(True).all(axis=axis, **kwargs).view(type(self))
+            r = self.filled(True).all(axis=axis, **kwargs)
+            if isinstance(r, bool):
+                d = type(self)(r)
+            else:
+                d = r.view(type(self))
```

Comment:
since `all` is `np.logical_and.reduce`, the `OO->O` mapping reduces the output to a boolean scalar, which has no view.

Reply:
Alternatively, could pass `keepdims=True`, and then throw away the dimension at the end